sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/AdtQueryBuilder.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text;
+using Azure.DigitalTwins.Core.QueryBuilder;
+
+namespace Azure.DigitalTwins.Core.Queries.QueryBuilders
+{
+    /// <summary>
+    /// Azure DigitalTwins Query builder that facilitates writing queries against ADT instances.
+    /// </summary>
+    public class AdtQueryBuilder
+    {
+        private readonly SelectQuery _selectQuery;
+        private readonly FromQuery _fromQuery;
+        private readonly WhereQuery _whereQuery;
+
+        /// <summary>
+        /// Initializes an instance of an AdtQueryBuilder object.
+        /// </summary>
+        public AdtQueryBuilder()
+        {
+            _whereQuery = new WhereQuery(this);
+            _fromQuery = new FromQuery(this, _whereQuery);
+            _selectQuery = new SelectQuery(this, _fromQuery);
+        }
+
+        /// <summary>
+        /// Specifies the list of columns that the query will return
+        /// </summary>
+        /// <param name="args"> The arguments that can be queried (eg *, somePropertyName, etc.) </param>
+        /// <returns> Query that contains a select clause. </returns>
+        public FromQuery Select(params string[] args)
+        {
+            return _selectQuery.Select(args);
+        }
+
+        /// <summary>
+        /// Used to return only a certain number of top items that meet the query requirements.
+        /// </summary>
+        /// <param name="count"> The argument for TOP(), i.e the number of instances to return. </param>
+        /// <param name="args"> The arguments that can be optionally passed with top (eg property name). </param>
+        /// <returns> Query that contains a select clause. </returns>
+        public FromQuery SelectTop(int count, params string[] args)
+        {
+            return _selectQuery.SelectTop(count, args);
+        }
+
+        /// <summary>
+        /// Used to return the number of items that meet the query requirements.
+        /// </summary>
+        /// <returns> Query that contains a select clause. </returns>
+        public FromQuery SelectCount()
+        {
+            return _selectQuery.SelectCount();
+        }
+
+        /// <summary>
+        /// Used to provide the entire SELECT clause as opposed to providing individual field (column) names.
```

Comment:
Not clear to me if, as a user, I need to include the `SELECT` keyword or not. Can you give an example in the docs to make that clear?

Reply:
I agree, I helped him with this summary comment and I think an example makes it a lot more clear!

Reply:
You can use this link as an example to add an example section:
https://github.com/Azure/azure-sdk-for-net/blob/dac0f7e6d518b16e5bded4c947dae3053baada72/sdk/digitaltwins/Azure.DigitalTwins.Core/src/DigitalTwinsClient.cs#L55

Reply:
Yup! There are samples in the Readme